### PR TITLE
fix: #451

### PR DESCRIPTION
**Pull request by eminfedar**

### CODE REVIEW

**Comment by Gama11** [comment]
Hm, a few thoughts:

- shouldn't the warning be printed to stderr instead of stdout?
- I'm not sure why the version would still be printed in case of too many parameters?
- I think the error message could be clearer, something like "no parameters expected, got <parameters>".

File: src/haxelib/client/Main.hx [hunk 1]
```diff
@@ -244,7 +244,10 @@ class Main {
 	}
 
 	function version() {
-		print(VERSION_LONG);
+		Sys.println('Haxelib\'s version: ${VERSION_LONG}');
+
+		if ( paramOpt() != null )
+			Sys.println('! WARNING: "version" parameter only shows the version of the "haxelib"');
```
Comment (RealyUniqueName) [inline src/haxelib/client/Main.hx]: That's a misleading error.
What if a user typed `haxelib version --verbose`?
The message should tell exactly, what's the problem. And the problem is invalid amound of arguments.

File: src/haxelib/client/Main.hx [hunk 2]
```diff
@@ -244,7 +244,10 @@ class Main {
 	}
 
 	function version() {
-		print(VERSION_LONG);
+		Sys.println('Haxelib\'s version: ${VERSION_LONG}');
```
Comment (RealyUniqueName) [inline src/haxelib/client/Main.hx]: It's better not to change output format for no reason. Some tools or CIs may rely on it.

File: src/haxelib/client/Main.hx (hunk 1 above)
Comment (eminfedar) [reply src/haxelib/client/Main.hx]: I can't get the difference with tag `--verbose`.

I think the only usage is `haxelib version`, isn't it?

Comment (RealyUniqueName) [reply src/haxelib/client/Main.hx]: It's just an example of a case when proposed error message would be misleading.

**Comment by eminfedar** [comment]
This is okay I think? -> feb175f

**Comment by RealyUniqueName** [comment]
That's better, but needs a newline in the end of the message and `Sys.exit(1)` after the message :)

**Comment by eminfedar** [comment]
Should `print(VERSION_LONG)` line exit with 0 too?

**Comment by RealyUniqueName** [comment]
No, that happens by default.

**Pull request merged**